### ISSUE TITLE
FSD EXP4

### ISSUE THREAD

**Issue by pramyaperiyasamy1234k-droid** [pramyaperiyasamy1234k-droid/HIRE#1]
<!DOCTYPE html>
<html ng-app="todoApp">
<head>
    <title>AngularJS To-Do List with Filters</title>
    <meta charset="UTF-8">
    <script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.8.2/angular.min.js"></script>

    <style>
        body {
            font-family: Arial, sans-serif;
            background: linear-gradient(to right, #667eea, #764ba2);
            margin: 0;
            padding: 0;
        }

        .container {
            width: 420px;
            margin: 60px auto;
            background: white;
            padding: 25px;
            border-radius: 10px;
            box-shadow: 0 8px 20px rgba(0,0,0,0.2);
        }

        h2 {
            text-align: center;
        }

        input[type="text"] {
            width: 68%;
            padding: 8px;
            margin-right: 5px;
        }

        button {
            padding: 8px 10px;
            border: none;
            cursor: pointer;
            border-radius: 5px;
            margin: 2px;
        }

        .add-btn {
            background: #4CAF50;
            color: white;
        }

        .delete-btn {
            background: #f44336;
            color: white;
            float: right;
        }

        .filter-btn {
            background: #2196F3;
            color: white;
        }

        .active-filter {
            background: #08ec7a;
        }

        ul {
            list-style: none;
            padding: 0;
        }

        li {
            padding: 10px;
            margin-top: 8px;
            background: #f1f1f1;
            border-radius: 5px;
        }

        .completed {
            text-decoration: line-through;
            color: gray;
            background: #d4edda;
        }

        .footer {
            margin-top: 15px;
            font-size: 14px;
            text-align: center;
            color: rgb(211, 100, 10);
        }

        .filters {
            text-align: center;
            margin-top: 10px;
        }
    </style>
</head>

<body ng-controller="TodoController">

<div class="container">
    <h2> Ramya P To-Do List</h2>

    <input type="text" ng-model="newTask" placeholder="Enter new task">
    <button class="add-btn" ng-click="addTask()">Add</button>

    <div class="filters">
        <button class="filter-btn" 
                ng-class="{'active-filter': filterType==='all'}"
                ng-click="setFilter('all')">All</button>

        <button class="filter-btn" 
                ng-class="{'active-filter': filterType==='active'}"
                ng-click="setFilter('active')">Active</button>

        <button class="filter-btn" 
                ng-class="{'active-filter': filterType==='completed'}"
                ng-click="setFilter('completed')">Completed</button>
    </div>

    <ul>
        <li ng-repeat="task in tasks | filter:taskFilter">
            <span ng-click="toggleTask(task)" 
                  ng-class="{'completed': task.completed}">
                {{task.name}}
            </span>
            <button class="delete-btn" ng-click="deleteTask($index)">X</button>
        </li>
    </ul>

    <div class="footer">
        Total: {{tasks.length}} |
        Active: {{activeCount()}} |
        Completed: {{completedCount()}}
    </div>
</div>

<script>
var app = angular.module("todoApp", []);

app.controller("TodoController", function($scope) {

    $scope.tasks = [
        { name: "Learn AngularJS", completed: false },
        { name: "Build To-Do App", completed: true }
    ];

    $scope.filterType = "all";

    $scope.addTask = function() {
        if ($scope.newTask && $scope.newTask.trim() !== "") {
            $scope.tasks.push({
                name: $scope.newTask,
                completed: false
            });
            $scope.newTask = "";
        }
    };

    $scope.deleteTask = function(index) {
        $scope.tasks.splice(index, 1);
    };

    $scope.toggleTask = function(task) {
        task.completed = !task.completed;
    };

    $scope.completedCount = function() {
        return $scope.tasks.filter(function(task) {
            return task.completed;
        }).length;
    };

    $scope.activeCount = function() {
        return $scope.tasks.filter(function(task) {
            return !task.completed;
        }).length;
    };

    $scope.setFilter = function(type) {
        $scope.filterType = type;
    };

    $scope.taskFilter = function(task) {
        if ($scope.filterType === "active") {
            return !task.completed;
        } else if ($scope.filterType === "completed") {
            return task.completed;
        }
        return true; // all
    };

});
</script>

</body>
</html>
